packages/dds/merge-tree/src/test/revertibleFarm.spec.ts

```diff
@@ -0,0 +1,200 @@
+                            }
+
+                            if (opsWithRevert > 0) {
+                                // add modifications from another client
+                                msgs.push(...generateOperationMessagesForClients(
+                                    mt,
+                                    seq,
+                                    [clients.A, clients.C],
+                                    logger,
+                                    opsWithRevert,
+                                    minLen,
+                                    defaultOptions.operations));
+                            }
+
+                            let redoBaseText: string | undefined;
+                            if (ackBeforeRevert !== "None") {
+                                const ackAll = ackBeforeRevert === "All";
+                                seq = applyMessages(
+                                    seq,
+                                    msgs.splice(
+                                        0,
+                                        ackAll
+                                            ? msgs.length
+                                            : random.integer(0, Math.floor(msgs.length / 2))(mt)),
+                                    clients.all,
+                                    logger);
+                                if (ackAll) {
+                                    redoBaseText = logger.validate({ errorPrefix: "Before Revert Ack" });
+                                }
+                            }
+
+                            try {
+                                revertMergeTreeDeltaRevertibles(clientBDriver, clientB_Revertibles.splice(0));
+                                seq = applyMessages(seq, msgs.splice(0), clients.all, logger);
+                            } catch (e) {
+                                throw logger.addLogsToError(e);
+                            }
+                            logger.validate({
+                                clear: true,
+                                baseText: opsWithRevert === 0 ? undoBaseText : undefined,
+                                errorPrefix: "After Revert (undo)",
+                            });
+
+                            try {
+                                // reset the callback before the final revert
+                                // to avoid accruing any new detached references
+                                clients.B.mergeTreeDeltaCallback = oldCallback;
+                                revertMergeTreeDeltaRevertibles(clientBDriver, clientB_Revertibles.splice(0));
+                                seq = applyMessages(seq, msgs.splice(0), clients.all, logger);
+
+                                walkAllChildSegments(
+                                    clients.B.mergeTree.root,
+                                    (seg: ISegment) => {
+                                        if (seg?.trackingCollection.empty === false) {
```

Comment:
what's the reason for the conditional here and below? does `assert.notDeepStrictEqual` not tolerate `undefined` or something? seems like you can get rid of the if statements

Reply:
so i can set break points before the assert happens

Reply:
fair--I prefer `assert.fail()` in that case if the assertion message doesn't give an informative error message (and doesn't seem like it would be particularly helpful here)